### PR TITLE
[ISSUE #5507]Improve the speed of AttributeParser#parseToMap parsing

### LINKED ISSUE #5507: Improve the speed of AttributeParser#parseToMap parsing
The issue tracker is used for bug reporting purposes **ONLY** whereas feature request needs to follow the [RIP process](https://github.com/apache/rocketmq/wiki/RocketMQ-Improvement-Proposal). To avoid unnecessary duplication, please check whether there is a previous issue before filing a new one.

It is recommended to start a discussion thread in the [mailing lists](http://rocketmq.apache.org/about/contact/) in cases of discussing your deployment plan, API clarification, and other non-bug-reporting issues.
We welcome any friendly suggestions, bug fixes, collaboration, and other improvements.

Please ensure that your bug report is clear and self-contained. Otherwise, it would take additional rounds of communication, thus more time, to understand the problem itself.

Generally, fixing an issue goes through the following steps:
1. Understand the issue reported;
1. Reproduce the unexpected behavior locally;
1. Perform root cause analysis to identify the underlying problem;
1. Create test cases to cover the identified problem;
1. Work out a solution to rectify the behavior and make the newly created test cases pass;
1. Make a pull request and go through peer review;

As a result, it would be very helpful yet challenging if you could provide an isolated project reproducing your reported issue. Anyway, please ensure your issue report is informative enough for the community to pick up. At a minimum, include the following hints:

**BUG REPORT**

1. Please describe the issue you observed:

image
split two times
```java
@BenchmarkMode(Mode.Throughput)
@Warmup(iterations = 3, time = 1)
@Measurement(iterations = 5, time = 5)
@Threads(1)
@Fork(1)
@State(value = Scope.Benchmark)
@OutputTimeUnit(TimeUnit.SECONDS)
public class SplitBenchmark {

    private String bbb = "1111,2222";

    @Benchmark
    public void splitTwo() {
        String s = bbb.split(",")[0];
        String s1 = bbb.split(",")[1];
    }

    @Benchmark
    public void splitOne() {
        String[] split = bbb.split(",");
        String s = split[0];
        String s1 = split[1];
    }

    public static void main(String[] args) throws RunnerException {
        Options opt = new OptionsBuilder()
            .include(SplitBenchmark.class.getSimpleName())
            .result("result.json")
            .resultFormat(ResultFormatType.JSON).build();
        new Runner(opt).run();
    }
}
```
benchmark result:
image

2. Please tell us about your environment:

3. Other information (e.g. detailed explanation, logs, related issues, suggestions on how to fix, etc):

**FEATURE REQUEST**

1. Please describe the feature you are requesting.

2. Provide any additional detail on your proposed use case for this feature.

2. Indicate the importance of this issue to you (blocker, must-have, should-have, nice-to-have). Are you currently using any workarounds to address this issue?

4. If there are some sub-tasks involved, use -[] for each sub-task and create a corresponding issue to map to the sub-task:

- [sub-task1-issue-number](example_sub_issue1_link_here): sub-task1 description here, 
- [sub-task2-issue-number](example_sub_issue2_link_here): sub-task2 description here,
- ...

### PR DESCRIPTION
**Pull request by mxsm**
**Make sure set the target branch to `develop`**

## What is the purpose of the change

close #5507 

## Brief changelog

- Improve the speed of AttributeParser#parseToMap parsing
- Polish code

## Verifying this change

XXXX

Follow this checklist to help us incorporate your contribution quickly and easily. Notice, `it would be helpful if you could finish the following 5 checklist(the last one is not necessary)before request the community to review your PR`.

- [x] Make sure there is a [Github issue](https://github.com/apache/rocketmq/issues) filed for the change (usually before you start working on it). Trivial changes like typos do not require a Github issue. Your pull request should address just this issue, without pulling in other changes - one PR resolves one issue. 
- [x] Format the pull request title like `[ISSUE #123] Fix UnknownException when host config not exist`. Each commit in the pull request should have a meaningful subject line and body.
- [x] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
- [x] Write necessary unit-test(over 80% coverage) to verify your logic correction, more mock a little better when cross module dependency exist. If the new feature or significant change is committed, please remember to add integration-test in [test module](https://github.com/apache/rocketmq/tree/master/test).
- [x] Run `mvn -B clean apache-rat:check findbugs:findbugs checkstyle:checkstyle` to make sure basic checks pass. Run `mvn clean install -DskipITs` to make sure unit-test pass. Run `mvn clean test-compile failsafe:integration-test`  to make sure integration-test pass.
- [ ] If this contribution is large, please file an [Apache Individual Contributor License Agreement](http://www.apache.org/licenses/#clas).

### CODE REVIEW

**Review by tsunghanjacktsai — APPROVED** [review]

File: common/src/main/java/org/apache/rocketmq/common/attribute/AttributeParser.java [hunk 1]
```diff
@@ -16,38 +16,44 @@
...
 
 public class AttributeParser {
+
+    public static final String COMMA = ",";
+
+    public static final String EQUAL_SIGN = "=";
+
+    public static final String PLUS_SIGN = "+";
+
+    private static final String MINUS_SIGN = "-";
```
Comment (caigy) [inline common/src/main/java/org/apache/rocketmq/common/attribute/AttributeParser.java:33]: Pls use more descriptive names, like 'ATTR_ARRAY_SEPARATOR'.

**Review by caigy — CHANGES_REQUESTED** [review]
Although those variables should be constant, are there any test results to show the exact effect in improving the speed of parsing?

File: common/src/main/java/org/apache/rocketmq/common/attribute/AttributeParser.java (hunk 1 above)
Comment (mxsm) [reply common/src/main/java/org/apache/rocketmq/common/attribute/AttributeParser.java:33]: > Pls use more descriptive names, like 'ATTR_ARRAY_SEPARATOR'.

I will do, thanks for you suggest

**Review by lizhanhui — DISMISSED** [review]

File: common/src/main/java/org/apache/rocketmq/common/attribute/AttributeParser.java [hunk 2]
```diff
@@ -16,38 +16,44 @@
...
 
 public class AttributeParser {
+
+    public static final String ATTR_ARRAY_SEPARATOR_COMMA = ",";
+
+    public static final String ATTR_ARRAY_SEPARATOR_EQUAL_SIGN = "=";
+
+    public static final String ATTR_ARRAY_SEPARATOR_PLUS_SIGN = "+";
+
+    private static final String ATTR_ARRAY_SEPARATOR_MINUS_SIGN = "-";
```
Comment (caigy) [inline common/src/main/java/org/apache/rocketmq/common/attribute/AttributeParser.java:33]: In fact not all of these constants are separators, pls give them names showing the exact purpose of them, eg. : What's the meaning of 'plus' or 'minus' in the context of attribute parsing?
BTW, I'd prefer making these constant fields as `private`, for they are not referenced outside `AttributeParse`.

**Review by caigy — CHANGES_REQUESTED** [review]

File: common/src/main/java/org/apache/rocketmq/common/attribute/AttributeParser.java (hunk 2 above)
Comment (mxsm) [reply common/src/main/java/org/apache/rocketmq/common/attribute/AttributeParser.java:33]: > In fact not all of these constants are separators, pls give them names showing the exact purpose of them, eg. : What's the meaning of 'plus' or 'minus' in the context of attribute parsing? BTW, I'd prefer making these constant fields as `private`, for they are not referenced outside `AttributeParse`.

I see what you mean, thanks for you suggest, I will polish this code again.

**Review by tsunghanjacktsai — APPROVED** [review]

**Review by caigy — APPROVED** [review]

**Review by RongtongJin — APPROVED** [review]

**Comment by mxsm** [comment]
@lizhanhui  PTAL~

**Pull request merged**